File: Python/BOJ/2606.py
N = int(input())
M = int(input())

v = [[] for i in range(N + 1)]
visited = [False for i in range(N + 1)]
for i in range(M):
    a, b = map(int, input().split())
    v[a].append(b)
    v[b].append(a)

visited[1] = True

cnt = 0
def recur(cur = 1):
    global cnt

    for i in v[cur]:
        if visited[i]:
            continue
        visited[i] = True
        cnt += 1
        recur(i)

recur()
print(cnt)

File: C++/BOJ/2606.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
#define MAX 101

bool visited[MAX];
vector<int> v[MAX];
int cnt = 0;

void dfs(int cur) {
    visited[cur] = true;
    cnt++;

    for (auto x: v[cur]) {
        if (visited[x]) continue;
        dfs(x);
    }
}

int main() {
    int N, M;
    cin >> N >> M;

    for (int i = 0; i < M; i++) {
        int a, b;
        cin >> a >> b;
        v[a].push_back(b);
        v[b].push_back(a);
    }

    dfs(1);

    cout << cnt - 1;
    return 0;
}